### PR TITLE
feat(suite): Send SOL

### PR DESCRIPTION
**Pull request by memoriesadrift**
PR is a mess because of rebases. Review last 4 commits 🙆‍♂️ 👍 
## Related Issue

Resolve

### CODE REVIEW

File: suite-common/wallet-utils/src/validation.ts [hunk 1]
```diff
@@ -29,7 +29,11 @@ const getCoinFromTestnet = (symbol: Account['symbol']) => {
 export const isAddressValid = (address: string, symbol: Account['symbol']) => {
     const networkType = getNetworkType(symbol);
     const updatedSymbol = getCoinFromTestnet(symbol);
-    return addressValidator.validate(address, updatedSymbol.toUpperCase(), networkType); // TODO(vl): validate address
+
+    // TODO(vl): validate address
```
Comment (memoriesadrift) [inline suite-common/wallet-utils/src/validation.ts]: Will revert this commit when [this PR](https://github.com/trezor/trezor-address-validator/pull/25) is merged into the address validator repo.

File: packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import { getExternalComposeOutput, getLamportsFromSol } from '@suite-common/wallet-utils';
+import BigNumber from 'bignumber.js';
+
+const FEE_PER_SIGNATURE = new BigNumber(500);
+
+// TODO(vl): Implement send-max
+// TODO(vl), phase 2: Implement custom Prioritization Fees
+
+// TODO(vl): Replace with appropriate call!
+const getRecentBlockhash = async () => {
```
Comment (memoriesadrift) [inline packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts]: Couldn't figure out how to add this to blockchain link, let's replace it with an appropriate call in an appropriate place :)

File: packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+        const { address, amount } = formValues.outputs[0];
+
+        // TODO(vl), phase 2: account for token transfers -- should only count fee when transferring tokens
+        const totalSpent = new BigNumber(amount)
+            .times(new BigNumber(10 ** 9))
+            .plus(FEE_PER_SIGNATURE)
+            .toString();
+
+        const feeObject = {
```
Comment (memoriesadrift) [inline packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts:32]: Fee calculation is not implemented in this PR

File: packages/blockchain-link/src/workers/solana/index.ts [hunk 4]
```diff
@@ -10,6 +10,16 @@ export type SolanaAPI = Connection;
 type Context = ContextType<SolanaAPI>;
 type Request<T> = T & Context;
 
+const pushTransaction = async (request: Request<MessageTypes.PushTransaction>) => {
```
Comment (PeterBenc) [inline packages/blockchain-link/src/workers/solana/index.ts:13]: Pls have you addressed feedback from [this PR](https://github.com/vacuumlabs/trezor-suite/pull/11)

File: packages/suite/package.json [hunk 5]
```diff
@@ -26,6 +26,7 @@
         "@hookform/resolvers": "3.1.0",
         "@reduxjs/toolkit": "1.9.5",
         "@sentry/core": "^7.56.0",
+        "@solana/web3.js": "^1.78.4",
```
Comment (PeterBenc) [inline packages/suite/package.json:29]: Lets put a pin in this, I suppose there should a way to include this dependency only in one module (its already in the blockchain link).

Can you please look into it?

File: suite-common/wallet-utils/src/sendFormUtils.ts [hunk 6]
```diff
@@ -474,3 +475,6 @@ export const getExcludedUtxos = ({
     });
     return excludedUtxos;
 };
+
+export const getLamportsFromSol = (amountInSol: string) =>
```
Comment (PeterBenc) [inline suite-common/wallet-utils/src/sendFormUtils.ts:479]: Correct me if I am wrong, but it seems that this util is used only in one place, and also, it seems there are not similar utils in this file. 

I would propose to move this util to `sendFormSolanaActions`

File: packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts (hunk 2 above)
Comment (PeterBenc) [reply packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts]: Recent blockhash should be in state, something  `getState().wallet.blockchain['solana']` but let me look further into this

**Review by PeterBenc — COMMENTED** [review]
So far looks good, sorry I havent had the time to test and go through this properly, I will check again on thursday.

File: packages/suite/package.json (hunk 5 above)
Comment (memoriesadrift) [reply packages/suite/package.json:29]: The linter complained about the dependency not being added, that's why I added it, I'll investigate 🤔

File: suite-common/wallet-utils/src/sendFormUtils.ts (hunk 6 above)
Comment (memoriesadrift) [reply suite-common/wallet-utils/src/sendFormUtils.ts:479]: The rationale for leaving it in here is that many chain specific utility functions used only in `sendForm<chain>Actions` live here for other blockchains too. You can ctrl+f `ETH` to find some.
In the [fee PR](https://github.com/vacuumlabs/trezor-suite/pull/15/files#diff-2a4187f6d5baecfcb050320f659168b1aa48eb9c6c8c46dfd67930d6cf19ab7e) I've further extended this.

**Comment by memoriesadrift** [comment]
Regarding the blockhash in `getState().wallet`, it seems to be outdated. How often does it get refetched?
Let's sync on this.

**Comment by memoriesadrift** [comment]
Closing in favour of https://github.com/vacuumlabs/trezor-suite/pull/16

**Pull request closed without merging**